content/en/agent/architecture.md

```diff
@@ -0,0 +1,112 @@
+
+On Windows the services are listed as:
+
+| Service               | Description           |
+|-----------------------|-----------------------|
+| DatadogAgent          | Datadog Agent         |
+| datadog-trace-agent   | Datadog Trace Agent   |
+| datadog-process-agent | Datadog Process Agent |
+
+By default the Agent binds 3 [ports][3] on Linux and 4 ports on Windows and OSX:
+
+| Port | Description                                                                                 |
+|------|---------------------------------------------------------------------------------------------|
+| 5000 | Exposes runtime metrics about the Agent.                                                    |
+| 5001 | Used by the Agent CLI and GUI to send commands and pull information from the running Agent. |
+| 5002 | Serves the GUI server on Windows and OSX.                                                   |
+| 8125 | Used for the DogStatsD server to receive external metrics.                                  |
+
+For information on configuring the ports, see [Network Traffic][4].
+
+### Collector
+
+The collector gathers all standard metrics every 15 seconds. Agent 6 embeds a Python 2.7 interpreter to run integrations and [custom checks][5].
+
+### Forwarder
+
+The Agent forwarder sends metrics over HTTPS to Datadog. Buffering prevents network splits from affecting metrics reporting. Metrics are buffered in memory until a limit in size or number of outstanding send requests is reached. Afterwards, the oldest metrics are discarded to keep the forwarder's memory footprint manageable. Logs are sent to Datadog over an SSL-encrypted TCP connection.
+
+### DogStatsD
+
+In Agent 6, DogStatsD is a Golang implementation of [Etsy's StatsD][6] metric aggregation daemon. DogStatsD receives and rolls up arbitrary metrics over UDP or a UNIX socket, allowing custom code to be instrumented without adding latency. Learn more about [DogStatsD][7].
+
+[1]: /metrics/custom_metrics/dogstatsd_metrics_submission/#metrics
+[2]: /tracing/guide/terminology/
+[3]: /agent/configuration/network/#open-ports
+[4]: /agent/configuration/network#configure-ports
+[5]: /developers/custom_checks/write_agent_check/
+[6]: https://github.com/etsy/statsd
+[7]: /metrics/custom_metrics/dogstatsd_metrics_submission/
+{{% /tab %}}
+{{% tab "Agent v5" %}}
+
+{{< img src="agent/agent5architecture.jpg" alt="Agent v5 Architecture" >}}
+
+Agent 5 is composed of four major components, each written in Python and running as a separate process:
+
+* **Collector** (`agent.py`): The collector runs checks on the current machine for configured [integrations][1], and captures system metrics, such as memory and CPU.
+* **DogStatsD** (`dogstatsd.py`): This is a StatsD-compatible backend server that you can send [custom metrics][2] to from your applications.
+* **Forwarder** (`ddagent.py`): The forwarder retrieves data from both DogStatsD and the collector, queues it up, and then sends it to Datadog.
```

Comment:
```suggestion
* **Forwarder** (`ddagent.py`): The forwarder retrieves data from both DogStatsD and the collector, queues it for submission, and then sends it to Datadog.
```
Just to avoid an idiom that might be confusing.